### PR TITLE
Feature/66

### PR DESCRIPTION
**Pull request by ddelpiano**
- Graph linked to json data.
- Hierarchy populated.
- refactored code to use Map
- other minor improvements

### CODE REVIEW

**Review by jrmartin — APPROVED** [review]
It looks good, the folder and files are showing under the Hierarchical View Component. The search functionality is broken, a different issue #73 will take care of it.

 I noticed that there's some nodes for contributors/persons and awards that are on the same level as the dataset node. 

For example, this dataset https://cassava.ucsd.edu/sparc/preview/archive/exports/2021-04-12T17%3A28%3A22%2C384227-07%3A00/datasets/N%3Adataset%3A02786240-2033-4f86-808d-daf345ce3165.ttl should have 12 contributors, according to this property : `TEMP:hasNumberOfContributors 12 ;`

There's 12 `sparc:DatasetContribution` objects in the turtle file, I think those are the ones that need to be displayed under contributors. Currently we only have 3, the other 9 are pixels away from the root node (screenshot). 
image


The 3 contributors added correctly, are the ones mentioned in this tag, not sure if coincidentally : 
```
        TEMP:hasContactPerson ORCID:0000-0002-1152-4884, 
        ORCID:0000-0002-8796-7614,
        ORCID:0000-0003-0787-1709 ;
```

**Comment by jrmartin** [comment]
The fix for the contributors worked @ddelpiano , it's looking good now 

I think I found an unrelated new glitch with this dataset https://cassava.ucsd.edu/sparc/preview/archive/exports/2021-04-12T17%3A28%3A22%2C384227-07%3A00/datasets/N%3Adataset%3A02d6f93c-56cd-471b-bbb9-99f65f47d203.ttl

 It's rendering the sparc:Sample nodes twice, or it looks like it is. I think what's happening is that 'sample:sub-9238_sub-9238_sam-1' from the turtle file is rendered as a node and directory 'sub-9238_sam-1' (same name as sample) is also rendered as a node. 
 In cases like this, where the graph has sparc:samples already with some folders and files, we don't need to display those same folders and files under the subject. We only display the folders and files under the subject if they haven't been added to a sparc:sample node in the graph, we can move this to a new issue if it's going to take a while to do these changes.
 image

File: src/utils/nodesFactory.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+function createImage(node) {
+    // TODO: replace this with a decorator (maybe).
+    const img = new Image();
+    img.src = (rdfTypes[String(node.type)]?.image !== "") ? rdfTypes[String(node.type)].image : rdfTypes.Uknown.image
+    return img;
+}
+
+function extractProperties(node, ttlTypes) {
+    for (const property of node.properties) {
```
Comment (zsinnema) [inline src/utils/nodesFactory.js:11]: recommendation: I would use and return a "new" node instead of updating the one passed in the function

**Review by zsinnema — APPROVED** [review]
lgtm

**Pull request merged**